third_party/2and3/cryptography/hazmat/primitives/asymmetric/ec.pyi

```diff
@@ -0,0 +1,112 @@
+from abc import ABCMeta, abstractmethod
+from typing import Union
+
+from cryptography.hazmat.backends.interfaces import EllipticCurveBackend
+from cryptography.hazmat.primitives.asymmetric.padding import AsymmetricPadding
+from cryptography.hazmat.primitives.asymmetric.utils import Prehashed
+from cryptography.hazmat.primitives.hashes import HashAlgorithm
+from cryptography.hazmat.primitives.serialization import Encoding, KeySerializationEncryption, PrivateFormat, PublicFormat
+from cryptography.x509 import ObjectIdentifier
+
+class EllipticCurve(metaclass=ABCMeta):
+    key_size: int
+    name: str
+
+class BrainpoolP256R1(EllipticCurve): ...
+class BrainpoolP384R1(EllipticCurve): ...
+class BrainpoolP512R1(EllipticCurve): ...
+class SECP192R1(EllipticCurve): ...
+class SECP224R1(EllipticCurve): ...
+class SECP256K1(EllipticCurve): ...
+class SECP256R1(EllipticCurve): ...
+class SECP384R1(EllipticCurve): ...
+class SECP521R1(EllipticCurve): ...
+class SECT163K1(EllipticCurve): ...
+class SECT163R2(EllipticCurve): ...
+class SECT233K1(EllipticCurve): ...
+class SECT233R1(EllipticCurve): ...
+class SECT283K1(EllipticCurve): ...
+class SECT283R1(EllipticCurve): ...
+class SECT409K1(EllipticCurve): ...
+class SECT409R1(EllipticCurve): ...
+class SECT571K1(EllipticCurve): ...
+class SECT571R1(EllipticCurve): ...
+
+class EllipticCurveOID:
+    SECP192R1: ObjectIdentifier
+    SECP224R1: ObjectIdentifier
+    SECP256K1: ObjectIdentifier
+    SECP256R1: ObjectIdentifier
+    SECP384R1: ObjectIdentifier
+    SECP521R1: ObjectIdentifier
+    BRAINPOOLP256R1: ObjectIdentifier
+    BRAINPOOLP384R1: ObjectIdentifier
+    BRAINPOOLP512R1: ObjectIdentifier
+    SECT163K1: ObjectIdentifier
+    SECT163R2: ObjectIdentifier
+    SECT233K1: ObjectIdentifier
+    SECT233R1: ObjectIdentifier
+    SECT283K1: ObjectIdentifier
+    SECT283R1: ObjectIdentifier
+    SECT409K1: ObjectIdentifier
+    SECT409R1: ObjectIdentifier
+    SECT571K1: ObjectIdentifier
+    SECT571R1: ObjectIdentifier
+
+class EllipticCurvePrivateKey(metaclass=ABCMeta):
+    key_size: int
+    @abstractmethod
+    def exchange(self, algorithm: ECDH, peer_public_key: EllipticCurvePublicKey) -> bytes: ...
+    @abstractmethod
+    def public_key(self) -> EllipticCurvePublicKey: ...
```

Comment:
This also has a `curve` abstract property.

Reply:
If it does, it doesn't seem to be documented: https://cryptography.io/en/latest/hazmat/primitives/asymmetric/ec/#cryptography.hazmat.primitives.asymmetric.ec.EllipticCurvePrivateKey

Reply:
I'll file an issue against `cryptography`, it's missing from the docs